static/app/components/globalSelectionLink.tsx

```diff
@@ -41,41 +42,41 @@ class GlobalSelectionLink extends React.Component<Props> {
       typeof to === 'object' && to.query ? {...globalQuery, ...to.query} : globalQuery;
 
     if (location) {
-      let toWithGlobalQuery;
+      let toWithGlobalQuery: LocationDescriptor;
+
       if (hasGlobalQuery) {
         if (typeof to === 'string') {
           toWithGlobalQuery = {pathname: to, query};
         } else {
           toWithGlobalQuery = {...to, query};
         }
       }
+
       const routerProps = hasGlobalQuery
-        ? {...this.props, to: toWithGlobalQuery}
+        ? {...this.props, to: toWithGlobalQuery!}
         : {...this.props, to};
 
-      return <RouterLink {...routerProps}>{this.props.children}</RouterLink>;
-    } else {
-      let queryStringObject = {};
-      if (typeof to === 'object' && to.search) {
-        queryStringObject = qs.parse(to.search);
-      }
+      return <Link {...routerProps}>{this.props.children}</Link>;
+    }
 
-      queryStringObject = {...queryStringObject, ...globalQuery};
+    let queryStringObject = {};
+    if (typeof to === 'object' && to.search) {
+      queryStringObject = qs.parse(to.search);
+    }
 
-      if (typeof to === 'object' && to.query) {
-        queryStringObject = {...queryStringObject, ...to.query};
-      }
+    queryStringObject = {...queryStringObject, ...globalQuery};
 
-      const url =
-        (typeof to === 'string' ? to : to.pathname) +
-        '?' +
-        qs.stringify(queryStringObject);
-      return (
-        <a {...this.props} href={url}>
-          {this.props.children}
-        </a>
-      );
+    if (typeof to === 'object' && to.query) {
+      queryStringObject = {...queryStringObject, ...to.query};
     }
+
+    const url =
+      (typeof to === 'string' ? to : to.pathname) + '?' + qs.stringify(queryStringObject);
+    return (
+      <a {...this.props} href={url}>
```

Comment:
should we use the `externalLink` component here?

Reply:
No because it is not necessarily an external link.

From the documentation at the top

>  * Falls back to <a> if there is no router present.